### ISSUE TITLE
Type inference of `defaultdict` fails when unpacking a `dict` with known value type

### ISSUE THREAD

**Issue by ringohoffman** [microsoft/pyright#9970]
**Describe the bug**

[One of `defaultdict`'s constructor overloads](https://github.com/python/typeshed/blob/af32625bd489f1f277ed984543bb87408b09d824/stdlib/collections/__init__.pyi#L404-L405) allows it to infer its value type from variadic keyword arguments. Pyright's inference of the value type works when passing in keyword arguments manually, but fails when unpacking a `dict` with a known value type.

**Code or Screenshots**

```python
import collections

dd_1 = collections.defaultdict(a=1)
reveal_type(dd_1)  # defaultdict[str, int]

init_dict = {"a": 1}
reveal_type(init_dict)  # dict[str, int]
dd_2 = collections.defaultdict(**init_dict)
reveal_type(dd_2)  # defaultdict[Unknown, Unknown]

```

[pyright playground](https://pyright-play.net/?code=JYWwDg9gTgLgBAYwgG2QUwTYEB2BnAKAIBNiB9ARjgF5EV1Nt8A6YtAMwEMBXZGY4JgAUnahQCUBKGgBuaTsjIwAnmDRDSlSQWA5gMMgMw04AbwBEncwC44FAL5TZ8xSrVDd%2Bw4JiTNAJhMkVAwsXDxWDh4%2BIxghACp4zwNYyWk5BSVVdQDxODgAYjg2Ll5%2BHwBtAFUcAGscCAB3HAAaOBr6ppwAXQIgA)

**Comment by erictraut** [comment]
This is due to an ambiguity in the typing spec regarding calls to an overloaded function when one or more arguments use an unpack operator. See [this recent thread](https://discuss.python.org/t/standardizing-argument-unpacking-behaviors-for-calls/80747/1) for details.

The `defaultdict` constructor includes multiple overloads, but there are two that are of interest here:

```python
    @overload
    def __init__(self) -> None: ...
    @overload
    def __init__(self: defaultdict[str, _VT], **kwargs: _VT) -> None: ...  # pyright: ignore[reportInvalidTypeVarUse]  #11780
```

The argument expression `**init_dict` arguably matches both of these overload signatures, since an unpack of a dict results in zero or more items. 

Since this is an open issue under active discussion, I'm going to leave this issue open for now. Once we come to a consensus on how this should be handled by type checkers, I'll make sure that pyright's behavior conforms.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.398.
